exporter/awscloudwatchlogsexporter/exporter.go

```diff
@@ -95,7 +96,7 @@ func newCwLogsExporter(config config.Exporter, params component.ExporterCreateSe
 }
 
 func (e *exporter) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
```

Comment:
If you have the opportunity to, I would encourage you to add some benchmarking for this function since I have concerns that trying to a get `pusher` for each `logEvent` isn't cheap.

Reply:
It has to be that way, other wise I would have to do batching and grouping in the exporter which I believe is against best practices?

Reply:
I am not sure what other way there is to do this.  Once a pusher is made it is stored in a cache so it should be quick.  The problem is the template for the names can change between entries, we need to check each time.

Reply:
> It has to be that way, other wise I would have to do batching and grouping in the exporter which I believe is against best practices?

What I expect the intent of this statement was that exporter should not queue input over multiple calls to send since this makes it hard to diagnose problems with what data is being lost and the collector can't retry the data since it is no longer in ownership of the pipeline.

> I am not sure what other way there is to do this. Once a pusher is made it is stored in a cache so it should be quick. The problem is the template for the names can change between entries, we need to check each time.

My assumption here is that you can do the following:
```
# Psuedo Code
ConsumeLogs(logs): 
   for entry in logs.Entries():
       cache.GetPusher(...).AddEntry(entry)
   cache.Flush()
```

What I am assuming here is that `AddEntry` will flush if the buffer is full but if it doesn't the forced flush after each entry has been passed would clear any existing buffers.